### PR TITLE
Cleaning up samples

### PR DESCRIPTION
**Pull request by t8y8**
Though not required, I moved all of the samples code to be wrapped in `if __name__ == "__main__"` blocks since that's best practice for command line scripts.

I also cleaned up imports to match pep8, and removed one or two that were unused.

I think we can refactor some of the really long lines a bit more to make them more readable -- especially with respect to the Filter/Sort options, but that can come in a later checkin.

### CODE REVIEW

File: samples/explore_datasource.py [hunk 1]
```diff
@@ -9,56 +9,61 @@
...
-import tableauserverclient as TSC
-import os.path
 import argparse
 import getpass
 import logging
 
+import tableauserverclient as TSC
+
+
 parser = argparse.ArgumentParser(description='Explore datasource functions supported by the Server API.')
```
Comment (RussTheAerialist) [inline samples/explore_datasource.py]: Should probably move these into the __main__ conditional also

File: samples/explore_datasource.py [hunk 2]
```diff
@@ -9,56 +9,61 @@
...
+    logging.basicConfig(level=logging_level)
 
-# SIGN IN
-tableau_auth = TSC.TableauAuth(args.username, password)
-server = TSC.Server(args.server)
-with server.auth.sign_in(tableau_auth):
-    # Query projects for use when demonstrating publishing and updating
-    all_projects, pagination_item = server.projects.get()
-    default_project = next((project for project in all_projects if project.is_default()), None)
+    # SIGN IN
```
Comment (RussTheAerialist) [inline samples/explore_datasource.py]: I'm not sure if this is just a product of github, but the spaces here are showing different than then ones below.  Make sure they are all spaces, not a mix of tabs and spaces.

File: samples/explore_workbook.py [hunk 3]
```diff
@@ -25,72 +26,78 @@
...
+            print([view.name for view in sample_workbook.views])
+
+            # Populate connections
+            server.workbooks.populate_connections(sample_workbook)
+            print("\nConnections for {}: ".format(sample_workbook.name))
+            print(["{0}({1})".format(connection.id, connection.datasource_name)
+                   for connection in sample_workbook.connections])
+
+            # Update tags and show_tabs flag
+            original_tag_set = set(sample_workbook.tags)
```
Comment (t8y8) [inline samples/explore_workbook.py]: This is a simpler way to copy a set and doesn't require copy.copy

File: samples/explore_datasource.py (hunk 2 above)
Comment (t8y8) [inline samples/explore_datasource.py]: Yeah I noticed that when it posted, weird, I'll make sure they're all spaces.

File: samples/explore_datasource.py (hunk 1 above)
Comment (t8y8) [inline samples/explore_datasource.py]: I debated that too -- I wonder if I should wrap everything in a main(), and then just have:

```
if __name__ == "__main__":
    main()
```

So it's easier to tell runtime from any setup?

**Pull request merged**